dax/processor_parser.py

```diff
@@ -942,16 +946,24 @@ def compare_to_existing(csesses, proc_type, parameter_matrix):
 
         assessors = [[] for _ in range(len(parameter_matrix))]
 
-        for casr in [a for a in csess.assessors() if a.type() == proc_type]:
-            inputs = casr.get_inputs()
-            if inputs is None:
-                LOGGER.warn('skipping, inputs field is empty:' + casr.label())
-                return list()
+        for casr in csess.assessors():
+            try:
+                proc_type_matches = (casr.type() == proc_type)
+            except:
+                LOGGER.error(f'Failed to check type of {casr.label()}')
+                continue
+            
+            if proc_type_matches:
+                inputs = casr.get_inputs()
+                if inputs is None:
+                    LOGGER.warn(f'Empty inputs - skipping {casr.label()}')
+                    continue
 
             for pi, p in enumerate(parameter_matrix):
                 if inputs == p:
-                    # BDB 6/5/21 do we ever have more than one assessor
-                    # with the same set of inputs?
+                    # BDB  6/5/21 do we ever have more than one assessor
+                    #             with the same set of inputs?
+                    # BPR 2/28/22 in principle we could
```

Comment:
how?

Reply:
Isn't this actually common? Two different T1 segmentation pipelines that only take the T1 as input, for example.

Reply:
If I am not understanding something, I may well have messed up that bit of the code so....

Reply:
But of the same proc type?

Reply:
I did mess it up - the pi,p loop should be inside the if proctype_match. New commit

Reply:
... or should it. Argh I don't understand the logic here

Reply:
I think it should. @bud42 can you verify the logic now is the same as it was before, just with added error catching?